replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ReplicatorHistoryImpl.java

```diff
@@ -134,8 +215,90 @@ private Metacard getMetacardFromStatus(ReplicationStatus replicationStatus) {
     helper.setIfPresent(mcard, ReplicationHistory.PULL_BYTES, replicationStatus.getPullBytes());
     helper.setIfPresent(mcard, ReplicationHistory.PUSH_BYTES, replicationStatus.getPushBytes());
     helper.setIfPresent(mcard, ReplicationHistory.STATUS, replicationStatus.getStatus().name());
+    helper.setIfPresent(mcard, ReplicationHistory.LAST_SUCCESS, replicationStatus.getLastSuccess());
+    helper.setIfPresent(mcard, ReplicationHistory.LAST_RUN, replicationStatus.getLastRun());
     mcard.setId(replicationStatus.getId());
     mcard.setTags(Collections.singleton(ReplicationHistory.METACARD_TAG));
     return mcard;
   }
+
+  private void addStats(ReplicationStatus base, ReplicationStatus status) {
+    base.setPushCount(base.getPushCount() + status.getPushCount());
+    base.setPushBytes(base.getPushBytes() + status.getPushBytes());
+    base.setPushFailCount(base.getPushFailCount() + status.getPushFailCount());
+    base.setPullCount(base.getPullCount() + status.getPullCount());
+    base.setPullBytes(base.getPullBytes() + status.getPullBytes());
+    base.setPullFailCount(base.getPullFailCount() + status.getPullFailCount());
+    if (base.getLastRun() == null || status.getStartTime().after(base.getLastRun())) {
+      base.setLastRun(status.getStartTime());
+    }
+    base.setStatus(status.getStatus());
+    if (base.getStartTime().after(status.getStartTime())) {
+      base.setStartTime(status.getStartTime());
+    }
+    if (status.getStatus().equals(Status.SUCCESS)) {
+      if (status.getPushCount() > 0 || status.getPullCount() > 0) {
+        base.setDuration(base.getDuration() + status.getDuration());
+      }
+      if (base.getLastSuccess() == null || status.getStartTime().after(base.getLastSuccess())) {
+        base.setLastSuccess(status.getStartTime());
+      }
+    }
+  }
+
+  /** */
+  void runResultsAsSystemUser() {
+    security.runAsAdmin(
+        () -> {
+          try {
+            security.runWithSubjectOrElevate(
+                () -> {
+                  getReplicationEvents()
+                      .stream()
+                      .map(ReplicationStatus::getReplicatorName)
+                      .distinct()
+                      .forEach(config -> condenseResults(getReplicationEvents(config)));
```

Comment:
```suggestion
                     .map(this::getReplicationEvents)
                     .forEach(this::condenseResults);
```

Reply:
❓ Are we not creating a new record each time we are condensing instead of creating one final one at the end of condensing all those we are iterating here?

Reply:
Yup. Updated